### DISCUSSION TITLE
DLL Parm setting

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#319] [Progress Updates]
The DLL call requires the setting of the type of all parameters. This will be achieved by a couple of new instructions:
- DLLDefine      defines a new dll parameter list 
- DLLSETINT     defines and pushes a single parameter of type INT to the parm list
- DLLSETINT     defines and pushes a single parameter of type STRING to the parm list   
- DLLSETFLOAT defines and pushes a single parameter of type FLOAT to the parm list

The parameter list will be used with the instruction CALLDLL 
The DLL has to be opened with OPENDLL

Instruction names are not fixed yet!

**Comment by adesutherland** [comment]
You mean DLLSETSTRING :-) but anyway sounds good. Also I can across this https://dyncall.org/index this may be a way to make things cross platform?

**Comment by rvjansen** [comment]
I wonder about DLL. These things are called this name on Windows (from OS/2 days actually), and more recently, they exist on z/OS with this name, where they are just one of many opportunities to link to code dynamicall. On Unices they are called shared object libraries, and on macOS Dylibs. It would be nice if we could use the same names on all platform (which will be no problem because they could be platformspecific - but then again dyncall. I would prefer something with LINK but I am not adamant that it needs to be changed. Just an inkling that DLL would be good if you are actually dynamically linking DLL's.

**Reply by adesutherland** [reply]
For us they are really external calls so maybe an "ext" prefix?
